### DISCUSSION TITLE
How to debug why an application is going out of sync.

### DISCUSSION THREAD

**Discussion by codex70** [argoproj/argo-cd#18241] [Q&A]
I'm deploying kafka with 3 pods using argocd and the bitnami chart. Everything works fine, but every few hours the kafka pods are getting destroyed and re-created. Nothing has changed on the specification, but it may be that something like one of the secrets has been changed automatically. When I look at the logs I can see that it shows kafka as being out of sync, but I can't what is out of sync.

Is there any log entry specifically that I should be looking for to help find the exact fields that are changing? It should be easy to ignore the specific fields once I work out where the problem is.

**Comment by ChristianCiach** [comment]
Couldn't you just disable auto-sync for a while until it goes out-of-sync again and then review the diff?

**Reply by codex70** [reply]
Great idea, will try and see if I can find the problem
